rust-bot/src/main.rs

```diff
@@ -98,14 +111,36 @@ fn create_channel(
 
     new_channel
 }
-
+#[derive(Debug)]
 struct Handler;
 
-impl EventHandler for Handler {}
-
+impl EventHandler for Handler {
+    #[instrument]
+    fn ready(&self, _: Context, _ready_info: serenity::model::gateway::Ready) {
+        info!(bot_is_ready = ?std::time::SystemTime::now());
+    }
+}
+#[instrument]
 fn main() {
     // Configure the client with your Discord bot token in the environment.
     let token = env::var("DISCORD_TOKEN").expect("Expected a token in the environment");
+    //Sets up the tracing stuff. 
+    let honeycomb_config = libhoney::Config {
+        options: libhoney::client::Options {
+            api_key: env::var("HONEYCOMB_API_KEY").expect("expected a honeycomb api key in the environment"),
+            dataset: env::var("HONEYCOMB_DATASET_NAME").expect("expected a honeycomb dataset name"),
+            ..libhoney::client::Options::default()
+        },
+        transmission_options: libhoney::transmission::Options::default(),
+    };
+
+    let honeycomb_tracing_layer = tracing_honeycomb::new_honeycomb_telemetry_layer("honeycomb-service", honeycomb_config);
```

Comment:
Does this seem right for a change to the string? it seems like a service name, which in this case is corgo-bot

```suggestion
    let honeycomb_tracing_layer = tracing_honeycomb::new_honeycomb_telemetry_layer("corgo-bot", honeycomb_config);
```

Reply:
A string would be fine. I did it as an environment variable since I was pushing it to my own honeycomb account. I can change it. Do you want both bots' telemetry together, or do you want me to name this one like `corgo-bot-rust`?

Reply:
corgo-bot-rust makes sense, yeah. I'll change the suggestion above

Reply:
Oh, you know what, I'm just reading this again. It looks like this is the name for the tracing layer and not the actual service.